### DISCUSSION TITLE
App breaks when php max children reached

### DISCUSSION THREAD

**Discussion by jacksbox** [nextcloud/all-in-one#3149] [Questions] [answered]
### Steps to reproduce
1. Use a slow server with 1 core (no also tested with 2 cores, same problem)
2. Open Photos or Memories app with lots of ungenerated previews
3. Scroll 
5. Nextcloud completely shuts down and does not recover

The issue seems to be, that all the preview generation request start new php processes with own database connections.
This will fast reach the max_children settings.
Guess that is ok, but after this limit is reached, it seems like the connections are never released and thus the database is blocked indefinitely.
Nextcloud is now not accessible anymore and the only way to get it running again is to restart the container. 
That seems not acceptable (badly formulated, sorry) - I think it needs to be made sure that it will always recover from errors like this.

### Expected behavior

Nextcloud should always recover from errors

### Actual behavior 

Nextcloud does not recover and needs to be restarted

### Host OS

Ubuntu 22.04

#### Nextcloud AIO version 

7.4.0

#### Current channel 

latest

#### Other valuable info 

Nextcloud container log
```
[09-Aug-2023 07:14:06] WARNING: [pool www] server reached max_children setting (38), consider raising it
```

DB container log:
```
...
2023-08-09 07:14:12.491 UTC [345] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.536 UTC [346] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.570 UTC [347] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.602 UTC [348] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.629 UTC [349] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.657 UTC [350] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.686 UTC [351] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.724 UTC [352] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.751 UTC [353] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.780 UTC [354] FATAL:  remaining connection slots are reserved for non-replication superuser connections
2023-08-09 07:14:12.821 UTC [355] FATAL:  remaining connection slots are reserved for non-replication superuser connections
```

**Comment by th0rgall** [comment]
I have the same issue. I'm trying out a default AiO install (with Nextcloud 27) on an old laptop with 4GB RAM and a dual-core processor (Celeron 2955U). After opening some of the sample files with the admin account on two different clients, my Nextcloud also got blocked in a matter of minutes.

My `max_children` count is 77, double the OP's 38 with one extra, which makes me guess that this limit is dynamically set according to the estimated power of the system (number of cores?).
```
[10-Aug-2023 01:44:35] WARNING: [pool www] server reached max_children setting (77), consider raising it
```
From a quick search, it seems that there are multiple related issues:
- https://github.com/nextcloud/docker/issues/1969
- https://github.com/nextcloud/server/issues/39063

It's probably worthwhile to play with the configuration values mentioned there, but I don't know directly where they should be configured in AiO.

**Comment by szaimen** [comment]
Hi, can you upgrade to v7.0.0 Beta and check if that resolves things? See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by jacksbox** [comment]
Nope, that does not really help - only change, nextcloud not completely shuts down forever, but recovers after 5min. 
But this still makes the photos and memories app unusable.

PS: a bit weird to close as completed without waiting if it is actually solved?

**Comment by szaimen** [comment]
How much ram do you have and which optional addons do you have enabled?

**Comment by jacksbox** [comment]
2GB, 2 CPUS

Only imaginary is active

**Reply by szaimen** [reply]
What if you disable imaginary? Does it still crash in that case?

**Reply by szaimen** [reply]
Also did you already upgrade to AIO v7.0.0?

**Reply by jacksbox** [reply]
I just upgraded to 4GB of RAM and the problem goes away...so I guess it is kinda solved, but maybe the min specs should be updated?

**Comment by th0rgall** [comment, marked as answer]
I've upgraded to the beta channel, AIO v7.0.0. I am running the Collabora, Imaginary and Fulltextsearch add-ons with the same 4GB RAM.

From some quick testing with a small upload batch, the issue seems to be resolved for me too! 👍

**Comment by chrda81** [comment]
I hit the same issue `WARNING: [pool www] server reached max_children setting (158), consider raising it` yesteray with AIO v7.0.0 and tried to figure out, how to set `pm.max_children` manually. I located the file /usr/local/etc/php-fpm.d/www.conf, but the value seems to be overridden by every restart. So I found the environment variable `PHP_MAX_CHILDREN` in the entrypoint script of the nextcloud container, but when I try to set this variable in the docker-compose.yml environment section of the master container it is still ignored and the calculated value of 158 is used.

Where do I have to set this environment variable or why is it still be ignored?

**Reply by szaimen** [reply]
HI, this is a bug of collabora and will be fixed with the next update of the richdocuments app.

**Reply by chrda81** [reply]
Do you have an approximate time frame for when that fix will happen?

**Reply by szaimen** [reply]
No. It depends on the @nextcloud/office team

**Reply by juliusknorr** [reply]
Not sure which fix you are referring to exactly? The pdf preview one?

We just released a new richdocuments batch of maintenance releases yesterday.

**Reply by szaimen** [reply]
Yes, the pdf preview generation problem. I see, thanks for the heads-up Julius! :)

@chrda81 can you try to upgrade the richdocuments app restart the server and check if this fixes your problems?

**Reply by szaimen** [reply]
@juliushaertl I am referring to https://github.com/nextcloud/richdocuments/pull/3146. Is it not part of https://github.com/nextcloud/richdocuments/releases/tag/v7.1.6?

**Reply by szaimen** [reply]
Ah apparently v9.x is for nc27? So there was not release yet?

**Reply by szaimen** [reply]
@chrda81 apparently the fix is not released yet but should be tomorrow.

**Reply by chrda81** [reply]
@szaimen What actions do I need to perform the update, if the fix is released? I checked the all-in-one master instance today and it still says that version 7.1.1 is the latest available release. In the NextCloud app section only 2 updates are visible: Contacts v.5.4.0 and Calendar v4.5.0.

**Reply by szaimen** [reply]
Yes, the update is still not released yet.

**Reply by chrda81** [reply]
Ok. I would be very grateful if you would leave some information here if the time comes. Thank you so much 😊

**Comment by ccoles146** [comment]
I have this same issue. Currently running v7.5.1 in Windows docker desktop. Core i3 (8th gen) with 8Gb RAM. Collabora and Imaginary are the only installed add-ons. It works fine with one or two people but if three or four (I'm not sure of the exact number) then it grinds to a halt and seems to require a restart to get going again.

**Reply by th0rgall** [reply]
I'm currently on AIO v7.11.2 (stable channel, NC27), so you seem several minor versions behind. I've also had the issue once more on 7.0.0 with my 8GB Raspberry Pi 4B; but it's been working well for a long time now. Can you try making a backup & updating AIO & your NC version?
